### PR TITLE
Dropwizard migrations

### PR DESCRIPTION
**Pull request by chr1sdi**
Introduce Flywaydb migration versioning and apply 1st migration script (`V2__Add_item_key_to_entry.sql`) to add `key` column to `entry` table.

**Friendly note:** the migration script is prefixed with V2 leaving a gap for future use of V1 for the schema creation script.

### CODE REVIEW

File: src/main/java/uk/gov/register/RegisterConfiguration.java [hunk 1]
```diff
@@ -72,10 +75,22 @@ public String getRegisterDomain() {
...
+    @JsonProperty
+    private FlywayFactory flywayFactory = new FlywayFactory();
+
     public DataSourceFactory getDatabase() {
         return database;
     }
 
+    public FlywayFactory getFlywayFactory() {
+        flywayFactory.setBaselineOnMigrate(true);
+        flywayFactory.setLocations(Arrays.asList("/sql"));
```
Comment (philandstuff) [inline src/main/java/uk/gov/register/RegisterConfiguration.java]: could use `Collections.singletonList()` here

File: src/main/java/uk/gov/register/RegisterConfiguration.java [hunk 2]
```diff
@@ -72,10 +75,22 @@ public String getRegisterDomain() {
...
+        flywayFactory.setPlaceholders(ImmutableMap.of("registerName", getRegisterName()));
```
Comment (philandstuff) [inline src/main/java/uk/gov/register/RegisterConfiguration.java]: Similarly `Collections.singletonMap()`

**Comment by chr1sdi** [comment]
:rage4: let's do it!

**Comment by philandstuff** [comment]
:yolo:

**Pull request merged**